foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/AbstractPomMigration.java

```diff
@@ -40,4 +53,54 @@ protected void detectAndSetIndent(File file) {
             throw new BatonException("Failed to get indent from POM:" + file, e);
         }
     }
+
+    /**
+     * Checks if a given pom model contains any dependencies with a provided group ID and artifact ID
+     */
+    protected boolean hasDependency(Model model, String groupId, String artifactId) {
+        return model.getDependencies()
+             .stream()
+             .anyMatch(dependency ->
+                 dependency.getGroupId().equals(groupId) &&
+                 dependency.getArtifactId().contains(artifactId)
+             );
+    }
+
+    /**
+     * Checks the dependency and dependency management section of a given pom model and all profiles for any matches
+     */
+    protected List<Dependency> getAllDependenciesByFilterForProject(Model model, Predicate<? super Dependency> filter) {
+        List<Dependency> matchingDependencies = this.getAllDependenciesByFilter(model, filter);
+        for (Profile profile : model.getProfiles()) {
+            matchingDependencies.addAll(this.getAllDependenciesByFilter(profile, filter));
+        }
+        return matchingDependencies;
+    }
+
+    /**
+     * Checks the dependency and dependency management section of a given pom model for any matches
+     */
+    protected List<Dependency> getAllDependenciesByFilter(ModelBase model, Predicate<? super Dependency> filter) {
```

Comment:
A:
```suggestion
    protected List<Dependency> getMatchingDependencies(ModelBase model, Predicate<? super Dependency> filter) {
```
and similarly `getMatchingDependenciesForProject`

Reply:
Updated